### PR TITLE
Remove project team ID detection

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [X] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *MINOR* [version update](https://semver.org/)

### Context

Pulls the latest go-xcode to with two improvements:
- https://github.com/bitrise-io/go-xcode/pull/146
- https://github.com/bitrise-io/go-xcode/pull/145

### Changes

- Update `go-jwt` and `go-xcode` dependency
- Handle breaking changes in `go-xcode`
- Add new input to define Apple team ID if needed

### CODE REVIEW

**Review by shams-ahmed — APPROVED** [review]

File: config.go [hunk 1]
```diff
@@ -51,35 +50,17 @@ func (c Config) ValidateCertificates() ([]string, []string, error) {
 	return pfxURLs, passphrases, nil
 }
 
-// Certificates returns an array of p12 file URLs and passphrases
-func (c Config) Certificates() ([]certdownloader.CertificateAndPassphrase, error) {
```
Comment (ofalvai) [inline config.go:55]: Replaced with existing implementation in `go-xcode`

File: go.sum [hunk 2]
```diff
@@ -19,6 +21,8 @@ github.com/dgrijalva/jwt-go v3.2.0+incompatible h1:7qlOGliEKZXTDg6OTjfoBKDXWrumC
 github.com/dgrijalva/jwt-go v3.2.0+incompatible/go.mod h1:E3ru+11k8xSBh+hMPgOLZmtrrCbhqsmaPHjLKYnJCaQ=
```
Comment (godrei) [inline go.sum:21]: why is not the old jwt removed from here?

Comment (ofalvai) [reply go.sum]: Thanks for spotting this, a `go mod tidy` solved it

**Review by godrei — DISMISSED** [review]

**Review by godrei — APPROVED** [review]

**Review by godrei — COMMENTED** [review]
Do not forget about the README update

**Review by godrei — APPROVED** [review]

**Pull request merged**